packages/graph/hash_graph/lib/graph/src/subgraph.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use type_system::uri::VersionedUri;
+use utoipa::{openapi, ToSchema};
+
+use crate::{
+    knowledge::{EntityId, KnowledgeGraphQueryDepth, PersistedEntity, PersistedLink},
+    ontology::{
+        OntologyQueryDepth, PersistedDataType, PersistedEntityType, PersistedLinkType,
+        PersistedPropertyType,
+    },
+};
+
+// TODO - This is temporary and introduced for consistency, we need to introduce actual IDs for
+//  links, should be revisited as part of https://app.asana.com/0/0/1203157172269853/f
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct LinkId {
+    source_entity_id: EntityId,
+    target_entity_id: EntityId,
+    #[schema(value_type = String)]
+    link_type_id: VersionedUri,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+#[serde(rename_all = "camelCase", untagged)]
+pub enum GraphElementIdentifier {
+    // TODO: can we create a new type just to generate a Utoipa line for VersionedURI and then use
+    //  that inside a `#[schema(value_type =` expression?
```

Comment:
Maybe we can write an addon for this? cc @thehabbos007
This would look similar to https://github.com/hashintel/hash/blob/97e2c283b9c19c8d0de97f68b95ac203d31c0fd8/packages/graph/hash_graph/lib/graph/src/api/rest/mod.rs#L172-L322

Reply:
We could have an addon, yes. We could also use a struct (with the name `VersionedUri)` and adding the component manually. Utoipa optimistically looks up the name, so even if we don't derive `ToSchema` on `VersionedUri`, it would try to resolve the name if we were to use it

Reply:
As expressed in another thread, I don't want to do that as part of this PR, adding a type with the same name would be pretty unclear when it comes to imports and such. If we do it, it should be encapsulated in a single PR so the reasoning is easy to find and we can revert if needs.

Reply:
Yes, agree. I didn't consider my comment as a requested change

Reply:
We're manually implementing the `ToSchema` anyway (I wrote the TODO before swapping to `untagged`) so I have removed the TODO. If we expose `VersionedUri` in other places we can investigate if we can display something nicer but I suspect OpenAPI limitations will not let us